dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,406 @@
+{
+    StepValue(false);
+}
+
+void NumberBox::OnSpinUpClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args)
+{
+    StepValue(true);
+}
+
+void NumberBox::OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args)
+{
+    switch (args.Key()) {
+        case winrt::VirtualKey::Up:
+        case winrt::VirtualKey::GamepadDPadUp:
+            StepValue(true);
+            break;
+        case winrt::VirtualKey::Down:
+        case winrt::VirtualKey::GamepadDPadDown:
+            StepValue(false);
+            break;
+    }
+}
+
+void NumberBox::OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args) {
+    if (HyperScrollEnabled()) {
+        int delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
+        if (delta > 0)
+        {
+            StepValue(true);
+        }
+        else if (delta < 0)
+        {
+            StepValue(false);
+        }
+    }
+}
+
+// Increments or decrements value by StepFrequency, wrapping if necessary
+void NumberBox::StepValue(bool sign)
+{
+    // Validating input before and after
+    ValidateInput();
+    double newVal = Value();
+
+    if (sign)
+    {
+        newVal += StepFrequency();
+    }
+    else
+    {
+        newVal -= StepFrequency();
+    }
+
+    // MinMaxMode Wrapping
+    if (MinMaxMode() == winrt::NumberBoxMinMaxMode::WrapEnabled && !IsInBounds(newVal))
+    {
+        while ( newVal > MaxValue() )
+        {
+            newVal = MinValue() + (newVal - MaxValue()) - 1;
+        }
+        while ( newVal < MinValue() )
+        {
+            newVal = MaxValue() - abs(newVal - MinValue()) + 1;
+        }
+        Value(newVal);
+        UpdateTextToValue();
+        return;
+    }
+
+    // Input Overwriting - Coerce to min or max
+    if (BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten && !IsInBounds(newVal) )
+    {
+        if (newVal > MaxValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MaxEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled) )
+        {
+            newVal = MaxValue();
+        }
+        else if (newVal < MinValue() && (MinMaxMode() == winrt::NumberBoxMinMaxMode::MinEnabled || MinMaxMode() == winrt::NumberBoxMinMaxMode::MinAndMaxEnabled))
+        {
+            newVal = MinValue();
+        }
+    }
+
+    // Update Text and Revalidate new value
+    Value(newVal);
+    UpdateTextToValue();
+    ValidateInput();
+}
+
+// Check if text resembles formulaic input to determine if parser should be executed
+bool NumberBox::IsFormulaic(winrt::hstring in)
+{
+    std::regex r("^([0-9()\\s]*[+-/*^]+[0-9()\\s]*)+$");
+    return (std::regex_match(winrt::to_string(in), r));
+}
+
+void NumberBox::EvaluateInput()
+{
+    double val;
+
+    try
+    {
+        val = NumberBoxParser::Compute(m_TextBox.Text());
+    }
```

Comment:
Exceptions should be used in exceptional cases. Especially since we're not doing anything with the return value, this should just return std::optional<double> and empty value means parsing failed.